api/src/main/java/org/openmrs/module/mirebalais/require/RequireUtil.java

```diff
@@ -35,6 +36,15 @@ public static String sessionLocationDoesNotHaveTag(LocationTagDescriptor descrip
         return new String("typeof sessionLocation !== 'undefined' && !hasMemberWithProperty(sessionLocation.tags, 'display','" + descriptor.name() + "')");
     };
 
+    // note that Java 8 Nashorn script engine support does not (fully?) support ECMAScript6 so we can't use arrow notation or map, etc
+    public static String visitHasEncounterOfType(EncounterTypeDescriptor descriptor) {
+        return new String("typeof visit !== 'undefined' && typeof visit.encounters !== 'undefined' && hasMemberThatEvaluatesTrue(visit.encounters, (function(encounter) { return encounter.encounterType.uuid === '" + descriptor.uuid() + "' } ))");
+    }
+
+    public static String visitDoesNotHaveEncounterOfType(EncounterTypeDescriptor descriptor) {
+        return new String("typeof visit !== 'undefined' && typeof visit.encounters === 'undefined' || !hasMemberThatEvaluatesTrue(visit.encounters, (function(encounter) { return encounter.encounterType.uuid === '" + descriptor.uuid() + "' } ))");
```

Comment:
```suggestion
        return new String("(visit && !visit.encounters) || " +
        "!visit.encounters.some(function(encounter) { " +
          "  return encounter.encounterType.uuid === '" + descriptor.uuid() +
          "' })");
```

Reply:
So I guess I'm not sure exactly what Nashorn supports, but it does not support "some" even if it is ECMA5... I basically tested a bunch of different methods in a unit test and nothing I wanted to use works.

I do agree with just changing the name of the method to "some" though... I was worried about naming collisions, but that's unlikely.  I will manually update the rest of this code and test.

Reply:
Sounds good, thanks